### ISSUE TITLE
Robot framework project path contain Chinese character throw exception

### ISSUE THREAD

**Issue by kenifanying** [nokia/RED#315]
I have a simple python library  and robot test suite on same directory, and the directory name contain Chinese character. The python library can be automatically discovered by RED but marked non-accessible. I tried to reload the library get some error:
```
Library specification generation problem

Problem occurred during generating libdoc for 'Telnet' library located at 'C:\Users\tony\robotframework\test子系统\Telnet.py'.
RED python session problem. Following exception has been thrown by python service:
            Traceback (most recent call last):
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\robot_session_server.py", line 42, in inner
                result['result'] = func(*args, **kwargs)
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\robot_session_server.py", line 91, in inner
                return to_call(*args, **kwargs)
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\robot_session_server.py", line 124, in inner
                return to_call(*args, **kwargs)
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\robot_session_server.py", line 64, in inner
                return func(*args, **kwargs)
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\robot_session_server.py", line 370, in create_libdoc_in_separate_process
                raise Exception(err.decode('UTF-8', 'replace'))
            Exception: Traceback (most recent call last):
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\red_libraries.py", line 122, in <module>
                python_paths = __decode_unicode_if_needed(sys.argv[3].split(";"))
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\red_libraries.py", line 111, in __decode_unicode_if_needed
                return [__decode_unicode_if_needed(elem) for elem in arg]
              File "C:\Users\tony\AppData\Local\Temp\RobotTempDir8198076229433796343\red_libraries.py", line 109, in __decode_unicode_if_needed
                return arg.decode('utf-8')
              File "C:\Python27\lib\encodings\utf_8.py", line 16, in decode
                return codecs.utf_8_decode(input, errors, True)
            UnicodeDecodeError: 'utf8' codec can't decode bytes in position 39-40: unexpected end of data
```
I upload the demo project `test子系统`, any assistance would be appreciated.

Python 2.7.16
Robot Framework 3.1.1 (Python 2.7.16 on win32)
RED - Robot Editor version 0.8.12
System: Windows 7 SP1 X86_64

[test子系统.zip](https://github.com/nokia/RED/files/3328471/test.zip)

**Comment by michalanglart** [comment]
This seem to be highly related to #316 

Similarly I cannot reproduce this specific exception.

Under python 2.7 in my case the library specification file is not generated too but it fails due to not being able to find `Telnet` module. In your case it is failing earlier when constructing the paths required to be added to `sys.path`.

Under python 3.7 it works for me.

**Comment by kenifanying** [comment]
> 
> 
> This seem to be highly related to #316
> 
> Similarly I cannot reproduce this specific exception.
> 
> Under python 2.7 in my case the library specification file is not generated too but it fails due to not being able to find `Telnet` module. In your case it is failing earlier when constructing the paths required to be added to `sys.path`.

Did you download `test子系统.zip`  and unzip it directly? I can reproduce this problem after add the  unzip directory `test子系统 ` by using "Open Projects from File System" menu. And it's ok if try to generate xml file using `python -m robot.libdoc test子系统\Telnet.py test.xml`.

![red_test1](https://user-images.githubusercontent.com/4479181/62679145-266ff980-b9e6-11e9-8057-3b26e6095e6c.png)


> 
> Under python 3.7 it works for me.

It works for me too under python 3.7.4.

**Comment by michalanglart** [comment]
Fixed in 0.9.0
